app/controllers/account/users_controller.rb

```diff
@@ -29,7 +29,8 @@ def new
   def create
     @user = User.new(resource_params)
     @user.assign_attributes(tenant: current_tenant, password: User::DEFAULT_PASSWORD)
-    if @user.save
+    if @user.valid? && legal_role?
```

Comment:
pay some more attention!!
role from params here is already assigned to `@user`

Reply:
my suggestion to make it simple
delete "role" param from  resource_params if user can't assign this role
+ add validation for role presence

so you don;t need to call ` legal_role? ` in controller in create/update actions

Reply:
also, it's better to check for role first before `#valid?` to avoid unnecessary callbacks triggered

Reply:
@igorkasyanchuk mb it'd be better to add `:created_by` column and make a direct validation since we have such dependency between creator and created user ?